### PR TITLE
Simplify definition of "readiness state"

### PR DESCRIPTION
**Pull request by jugglinmike**
The definition of "maximum active sessions" states:

> [...] must be exactly one for a remote end that is an endpoint node.

This restriction is further supported by the definition of "active
sessions," which states:

> A remote end that is not an intermediary node has at most one active
> session at a given time.

Therefore, the "readiness state" for endpoint nodes can be described in
terms of a precise number ("one") of active sessions. The additional
state concerning the current user prompt is subsumed by this condition.

Simplify the definition of "readiness state" to reflect the limitations
of endpoint nodes.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/webdriver/883)

### CODE REVIEW

**Comment by shs96c** [comment]
Your PR highlighted an issue with the way that `readiness state` is calculated for intermediary nodes --- they never took into account the number of `active sessions` (raised as #891) This means that we should probably clarify how we define this condition, to possibly be a logical AND on

# False if the maximum active sessions is equal to the length of the list of active sessions. Otherwise true.
# False if the node is an intermediary node and is known to be in a state in which attempting to create a new session would fail. Otherwise true.
# False is the node is an endpoint node and there is a current user prompt present. Otherwise true.

We can either modify this PR or create a new one.

**Comment by jugglinmike** [comment]
I'm happy to follow up on this, but I believe that collapsing the conditions
across node types makes the endpoint-node-specific condition superfluous.
Basically, if "the node is an endpoint node and there is a current user prompt
present," then this implies that "the maximum active sessions is equal to the
length of the list of active sessions." This is because "the maximum active
sessions [...] must be exactly one for a remote end that is an endpoint node."

In other words, I believe this could be simplified to a logical AND on:

- False if the maximum active sessions is equal to the length of the list of
  active sessions. Otherwise true.
- False if the node is an intermediary node and is known to be in a state in
  which attempting to create a new session would fail. Otherwise true.

**Comment by shs96c** [comment]
You are correct. We can collapse this down to those two conditions. If we can phrase this using positive language, I think it'll be slightly clearer, but I could be wrong --- I'll trust your judgement.

**Comment by jugglinmike** [comment]
I've had to amend the commit message and force-push, so I've made the original
state of this branch available here, in case that's useful:

> If we can phrase this using positive language, I think it'll be slightly
> clearer, but I could be wrong --- I'll trust your judgement.

If you prefer, this would look like:

> A remote end's readiness state is True if all of the following conditions are
> met:
>
> - the maximum active sessions is equal to the length of the list of active
>   sessions
> - the node is an endpoint node or is known to be in a state in which
>   attempting to create a new session would succeed

Of course, the other variable in this is whether to use the "switch"
construction or a list of conditions. Happy to re-phrase as you prefer!

**Comment by shs96c** [comment]
Thanks for fixing this and bearing with the reviews!

**Comment by jugglinmike** [comment]
Happy to help out :)
